core/src/kmx/kmx_plus.cpp

```diff
@@ -158,6 +159,24 @@ COMP_KMXPLUS_KEYS::valid(KMX_DWORD _kmn_unused(length)) const {
   return true;
 }
 
+std::u16string
+COMP_KMXPLUS_KEYS_ENTRY::get_string() const {
+  if (flags & LDML_KEYS_FLAGS_EXTEND) {
+    return std::u16string();  // error, extended
+  }
```

Comment:
If this is an error, we need an assert() on it for debug builds. And I think the function should be named `unextended_to_as_string` or something like that?

Otherwise, this needs a TODO-LDML on it?

Reply:
Zero length is unambiguous- no bit pattern would produce it.

Reply:
In my understanding, we shouldn't be calling this with a non-extend keys entry, so it should be fixed in caller -- that's why I wanted the assert. Empty string may cause all sorts of other issues on a release build where caller is expecting non-empty string?

Reply:
changed to assert.